pkg/controller/enterprisesearch/config.go

```diff
@@ -135,3 +148,46 @@ func tlsConfig(ents entsv1beta1.EnterpriseSearch) *settings.CanonicalConfig {
 		"ent_search.ssl.certificate_authorities": []string{filepath.Join(certsDir, certificates.CAFileName)},
 	})
 }
+
+// configRefWatchName returns the name of the watch registered on
+func configRefWatchName(ents types.NamespacedName) string {
+	return fmt.Sprintf("%s-%s-configref", ents.Namespace, ents.Name)
+}
+
+// parseConfigRef builds a single merged CanonicalConfig from the secrets referenced in configReg,
+// and ensures watches are correctly set on those secrets.
+func parseConfigRef(c k8s.Client, dynamicWatches watches.DynamicWatches, ents entsv1beta1.EnterpriseSearch) (*settings.CanonicalConfig, error) {
+	cfg := settings.NewCanonicalConfig()
+	secretNames := make([]string, 0, len(ents.Spec.ConfigRef))
+	for _, secretRef := range ents.Spec.ConfigRef {
+		if secretRef.SecretName == "" {
+			continue
+		}
+		secretNames = append(secretNames, secretRef.SecretName)
+	}
+	nsn := k8s.ExtractNamespacedName(&ents)
+	if err := watches.WatchUserProvidedSecrets(nsn, dynamicWatches, configRefWatchName(nsn), secretNames); err != nil {
+		return nil, err
+	}
+	for _, secretName := range secretNames {
+		var secret corev1.Secret
+		if err := c.Get(types.NamespacedName{Namespace: ents.Namespace, Name: secretName}, &secret); err != nil {
+			// the secret may not exist (yet) in the cache
+			// it may contain important settings such as encryption keys, that we don't want to generate ourselves
+			// let's explicitly error out
+			return nil, err
+		}
+		if data, exists := secret.Data[ConfigFilename]; exists {
+			parsed, err := settings.ParseConfig(data)
+			if err != nil {
+				log.Error(err, "unable to parse configuration from secret",
```

Comment:
Should we also create an event here to raise visibility in case of misconfiguration?

Reply:
Good idea!